### PR TITLE
Revoke One Time Token resource 

### PR DESCRIPTION
**Pull request by GihanAyesh**
### Purpose
Fixes https://github.com/wso2/api-manager/issues/254

The [Design Documentation](https://docs.google.com/document/d/1gfhI9ozfHf8qV4SWRKfeUWFW8kSs0zAV2XFPnXGGiI0/edit?usp=sharing)

Please do not merge without the approval of @tharindu1st @HiranyaKavishani

### CODE REVIEW

File: components/wso2is.key.manager.operations.endpoint/src/test/java/org/wso2/is/key/manager/operations/endpoint/impl/RevokeOneTimeTokenApiServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (tharindu1st) [inline components/wso2is.key.manager.operations.endpoint/src/test/java/org/wso2/is/key/manager/operations/endpoint/impl/RevokeOneTimeTokenApiServiceImplTest.java:2]: Inc -> LLC

**Review by tharindu1st — APPROVED** [review]

File: components/wso2is.key.manager.operations.endpoint/src/main/java/org/wso2/is/key/manager/operations/endpoint/impl/RevokeOneTimeTokenApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+     *
+     * @param revokeTokenInfo {@link RevokeTokenInfoDTO}  with JWT token and consumer key
+     * @return response whether the request is completed or any other issue
+     */
+    public Response revokeOneTimeTokenPost(RevokeTokenInfoDTO revokeTokenInfo, MessageContext messageContext) {
+
+        String token = revokeTokenInfo.getToken();
+        String consumerKey = revokeTokenInfo.getConsumerKey();
+
+        if (StringUtils.isEmpty(token)) {
```
Comment (HiranyaKavishani) [inline components/wso2is.key.manager.operations.endpoint/src/main/java/org/wso2/is/key/manager/operations/endpoint/impl/RevokeOneTimeTokenApiServiceImpl.java:51]: What happens when token is a string with a spaces?

File: components/wso2is.key.manager.operations.endpoint/src/main/java/org/wso2/is/key/manager/operations/endpoint/impl/RevokeOneTimeTokenApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity(UserInfoUtil.getError(Response.Status.BAD_REQUEST.toString(),
+                            "Could not revoke the token because the token id is empty", null
+                    ))
+                    .build();
+        }
+        if (StringUtils.isEmpty(consumerKey)) {
```
Comment (HiranyaKavishani) [inline components/wso2is.key.manager.operations.endpoint/src/main/java/org/wso2/is/key/manager/operations/endpoint/impl/RevokeOneTimeTokenApiServiceImpl.java:58]: same here

File: components/wso2is.key.manager.operations.endpoint/src/gen/java/org/wso2/is/key/manager/operations/endpoint/dto/RevokeTokenInfoDTO.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
+package org.wso2.is.key.manager.operations.endpoint.dto;
```
Comment (HiranyaKavishani) [inline components/wso2is.key.manager.operations.endpoint/src/gen/java/org/wso2/is/key/manager/operations/endpoint/dto/RevokeTokenInfoDTO.java:1]: Missing license header

File: components/wso2is.key.manager.operations.endpoint/src/gen/java/org/wso2/is/key/manager/operations/endpoint/RevokeOneTimeTokenApiService.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package org.wso2.is.key.manager.operations.endpoint;
```
Comment (HiranyaKavishani) [inline components/wso2is.key.manager.operations.endpoint/src/gen/java/org/wso2/is/key/manager/operations/endpoint/RevokeOneTimeTokenApiService.java:1]: Missing license header

File: components/wso2is.key.manager.operations.endpoint/src/gen/java/org/wso2/is/key/manager/operations/endpoint/RevokeOneTimeTokenApi.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package org.wso2.is.key.manager.operations.endpoint;
```
Comment (HiranyaKavishani) [inline components/wso2is.key.manager.operations.endpoint/src/gen/java/org/wso2/is/key/manager/operations/endpoint/RevokeOneTimeTokenApi.java:1]: Missing license header

**Pull request merged**